src/client/scripts/game/chess/organizedlines.js

```diff
@@ -24,342 +24,358 @@ import coordutil from '../misc/coordutil.js';
+ * which is the nearest x-value the line intersects on or after the y-axis.
+ * For example, the line with step-size [2,0] that starts on point (0,0) will have an X value of '0',
+ * whereas the line with step-size [2,0] that starts on point (1,0) will have an X value of '1',
+ * because it's step size means it never intersects the y-axis at x = 0, but x = 1 is the nearest it gets to it, after 0.
+ * 
+ * If the line is perfectly vertical, the axis will be flipped, so `X` in this
+ * situation would be the nearest **Y**-value the line intersects on or above the x-axis.
+ * @param {number[]} step - [dx,dy]
+ * @param {number[]} coords - Coordinates that are on the line
+ * @returns {number} The X in the line's key: `C|X`
+ */
+function getXFromLine(step, coords) {
+    // See these desmos graphs for inspiration for finding what line the coords are on:
+    // https://www.desmos.com/calculator/d0uf1sqipn
+    // https://www.desmos.com/calculator/t9wkt3kbfo
+
+    const lineIsVertical = step[0] === 0;
+    const deltaAxis = lineIsVertical ? step[1] : step[0];
+    const coordAxis = lineIsVertical ? coords[1] : coords[0];
+    return math.posMod(coordAxis, deltaAxis);
+}
+
+/**
+ * Tests if the provided gamefile has colinear organized lines present in the game.
+ * This can occur if there are sliders that can move in the same exact direction as others.
+ * For example, [2,0] and [3,0]. We typically like to know this information because
+ * we want to avoid having trouble with calculating legal moves surrounding discovered attacks
+ * by using royalcapture instead of checkmate.
+ * @param {gamefile} gamefile 
+ */
+function areColinearSlidesPresentInGame(gamefile) {
+    const slidingPossible = gamefile.startSnapshot.slidingPossible; // [[1,1],[1,0]]
+
+    // How to know if 2 lines are colinear?
+    // They will have the exact same slope!
+
+    // Iterate through each line, comparing its slope with every other line
+    for (let a = 0; a < slidingPossible.length - 1; a++) {
+        const line1 = slidingPossible[a]; // [dx,dy]
+        const slope1 = line1[1] / line1[0]; // Rise/Run
+        const line1IsVertical = isNaN(slope1);
+        
+        for (let b = a + 1; b < slidingPossible.length; b++) {
+            const line2 = slidingPossible[b]; // [dx,dy]
+            const slope2 = line2[1] / line2[0]; // Rise/Run
+            const line2IsVertical = isNaN(slope2);
+
+            if (line1IsVertical && line2IsVertical) return true; // Colinear!
+            if (slope1 === slope2) return true; // Colinear!
+        }
+    }
+    return false;
+}
+
+export default {
+    initOrganizedPieceLists,
+    resetOrganizedLists,
+    organizePiece,
+    removeOrganizedPiece,
+    initUndefineds,
+    appendUndefineds,
+    areWeShortOnUndefineds,
+    addMoreUndefineds,
+    insertUndefinedIntoList,
+    buildKeyListFromState,
+    buildStateFromKeyList,
+    getEmptyTypeState,
```

Comment:
getEmptyTypeState() will only ever be called from inside this module, so it doesn't need to be exported.